### PR TITLE
Gvf for rotorcrafts

### PR DESCRIPTION
**Pull request by spacewolfXfr**
Add support in standard (non-parametric) GVF for rotorcrafts, by setting the `nav` variables appropriately.
Additionally, setup acceleration command in generic guidance for rotorcrafts.

### CODE REVIEW

**Review by noether — DISMISSED** [review]

File: sw/airborne/modules/guidance/gvf/gvf.c [hunk 1]
```diff
@@ -184,12 +185,34 @@ void gvf_control_2D(float ke, float kn, float e,
...
+
+  nav.speed.x = pdx_dot;
+  nav.speed.y = pdy_dot;
+
+  nav.accel.x = pd_dot_dot_x * kn;
+  nav.accel.y = pd_dot_dot_y * kn;
+
+  nav.heading = atan2f(pdx_dot,pdy_dot);
+
+  // I don't know how gvf_common works ... so I use nav directly
```
[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply sw/airborne/modules/guidance/gvf/gvf.c:206]: I would like to get rid of the ABI hack at least in the case of nav. The new guidance API is designed to implement direct acceleration control (for the controller that supports it at least). So I'm fine with this. The final would be to remove the gvf_low_level and let the controllers decide how they apply the requested setpoints.

**Review by gautierhattenberger — DISMISSED** [review]
Can be merged if tested in flight

File: sw/airborne/modules/guidance/gvf/gvf.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (noether) [reply sw/airborne/modules/guidance/gvf/gvf.c:206]: Ah, I see, cool. Then, there is no ABI regarding guidance anymore, and we must use the corresponding functions to set the desired velocity/acceleration to eliminate gvf_low_level.

**Comment by spacewolfXfr** [comment]
- Added the full chain from navigation acceleration to command to the controller using the standardized `guidance_h`,`guidance_v` interface.
- Commented out WIP acceleration command in GVF, as not tested in flight nor proved
- Synchronized changed made online (clearing out some comments)

File: sw/airborne/firmwares/rotorcraft/guidance/guidance_h.c [hunk 2]
```diff
@@ -605,6 +613,8 @@ void guidance_h_set_body_acc(float ax, float ay)
 void guidance_h_set_acc(float ax, float ay)
 {
   SetBit(guidance_h.sp.mask, 4);
+  ClearBit(guidance_h.sp.mask, 5);
```
[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply sw/airborne/firmwares/rotorcraft/guidance/guidance_h.c]: replaced by enum

File: sw/airborne/modules/guidance/gvf/gvf.c [hunk 3]
```diff
@@ -177,34 +177,52 @@ void gvf_control_2D(float ke, float kn, float e,
...
+  nav.speed.x = md_x * kn;
+  nav.speed.y = md_y * kn;
 
-  #ifdef ROTORCRAFT_FIRMWARE
 
-  // Speed normalization
-  pdx_dot *= (kn/norm_pd_dot);
-  pdy_dot *= (kn/norm_pd_dot);
+  // Acceleration induced by the field with speed set to kn (!WIP! Disabled for now)
+  #if 0
```
[the remark this answers is not in the document]
Comment (gautierhattenberger) [reply sw/airborne/modules/guidance/gvf/gvf.c]: enabled with warning

**Review by spacewolfXfr — COMMENTED** [review]
Validated in simulation (the aircraft no longer overshoot when switching between guidance modes)

**Pull request merged**